csrc/fusion_segmenter.cpp

```diff
@@ -1883,7 +1883,10 @@ void eraseInputDistinctRootDomains(Fusion* fusion) {
       std::vector<IterDomain*> new_alloc;
       new_alloc.reserve(tv->getAllocationDomain().size());
       for (IterDomain* alloc_id : tv->getAllocationDomain()) {
-        new_alloc.push_back(replay.getReplay().at(alloc_id));
+        IterDomain* new_alloc_id = replay.getReplay().at(alloc_id);
+        // FIXME: should this be taken care of by ReplayTransformations?
```

Comment:
Typically, propagation of parallel types is done with `scheduler_utils::parallelizeAllLike`. I feel like `ReplayTransformations` should just propagate transformations, but that's likely in part because I'm used to it.